packages/frontend/src/pages/Dashboard.tsx

```diff
@@ -556,14 +556,24 @@ const Dashboard: FC = () => {
         (c) => c.hasTilesThatSupportFilters,
     );
 
+    if (isDashboardLoading) {
+        return <PageSpinner />;
+    }
+
     if (dashboardError) {
         return <ErrorState error={dashboardError.error} />;
     }
-    if (dashboard === undefined) {
+
+    if (!dashboard) {
         return (
-            <Box mt="md">
-                <SuboptimalState title="Loading..." loading />
-            </Box>
+            <ErrorState
+                error={{
+                    name: 'NotExistsError',
+                    statusCode: 404,
+                    message: 'Dashboard not found',
+                    data: {},
+                }}
+            />
```

Comment:
![CleanShot 2025-12-25 at 15.45.51@2x.png](https://app.graphite.com/user-attachments/assets/412226f3-ddbb-4923-a830-4640f151338f.png)

when dashboard is not found. should this say Not found error?